File: BOJ/python3/2231.py
'''
분해합
https://www.acmicpc.net/problem/2231
'''
import sys
import time
start = time.time()  # 시작 시간
sys.stdin = open("input.txt", "r")  # input.txt 파일로 입력

N = int(input())

for M in range(1, N+1):
    # M에서의 분해합 계산
    n = M
    sum_N = n
    while n != 0:
        sum_N = sum_N + (n % 10)

        n //= 10

    # M이 N의 생성자면 출력
    if sum_N == N:
        print(M)
        break
else:
    # 생성자가 없는 경우 0 출력
    print(0)


# 현재시각 - 시작시간 = 실행 시간(sec)
print("--------------------")
print(f"Elapsed Time: {round(time.time() - start, 3)}s")

File: BOJ/nodejs/2231.js
/*
분해합
boj.kr/2231
 */
const start = new Date().getTime();

const fs = require("fs");
const input = fs.readFileSync("input.txt").toString().split("\n");
//const input = fs.readFileSync("/dev/stdin").toString().split("\n");

const N = 1 * input[0];

let ans = 0;
for (let m = 1; m < N; m++) {
  const M =
    String(m)
      .split("")
      .map(Number)
      .reduce((acc, cur) => acc + cur, 0) + m;

  if (M === N) {
    ans = m;
    break;
  }
}

console.log(ans);

const end = new Date().getTime();
console.log("--------------------");
console.log(`Elapsed Time: ${(new Date().getTime() - start) / 1000}s`);
